tools/webidl_binder.py

```diff
@@ -382,7 +382,7 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
       call_prefix += '!!('
       call_postfix += ')'
 
-  args = ['arg%d' % i for i in range(max_args)]
+  args = [(all_args[i].identifier.name if isinstance(all_args[i], WebIDL.IDLArgument) else ('arg%d' % i)) for i in range(max_args)]
```

Comment:
Silly question: when is an argument not an IDLArgument?

I think the naming restriction is a bit unfortunate, fwiw.  Consider just adding a loop to append `_` to names until they are unique.

Reply:
This is the case for automatically generated functions. For example if you define something like this in idl:
```
interface SomeInfo {
  attribute long someAttribute;
};
```
it will automatically create a getter and setter, like:
```
int EMSCRIPTEN_KEEPALIVE emscripten_bind_SomeInfo_get_someAttribute_0(SomeInfo* self) {
  return self->someAttribute;
}
void EMSCRIPTEN_KEEPALIVE emscripten_bind_SomeInfo_set_someAttribute_1(SomeInfo* self, int arg0) {
  self->someAttribute = arg0;
}
```
So basically automatically generated functions keep automatically generated argument names.

Reply:
Oh, and to answer the second part, the only additional restriction is they you can't name an argument 'self', since that one's already taken. 'self' is a very common keyword in languages anyway (Rust, Python, Objective-C, Swift for example), so I don't think this is unreasonable.